File: v8_liskov_substitution_principle/java/Main4.java
package v8_liskov_substitution_principle.java;

public class Main4 {

    public static void main(String[] args) {
        Calculator c1 = new Calculator();
        int x = 1;
        int y = 2;
        int addResult = c1.calculate(x, y);
        System.out.println(addResult);

        SuperCalculator c2 = new SuperCalculator();
        int subtractResult = c2.subtract(x, y);
        System.out.println(subtractResult);
    }
}

class Calculator {
	public int calculate(int n1, int n2) { return n1 + n2; }
}

class SuperCalculator extends Calculator {
	public int subtract(int n1, int n2) { return n1 - n2;}
}

File: v8_liskov_substitution_principle/python/main4.py
class Calculator:

    def calculate(self, n1, n2):
        return n1 + n2

class SuperCalculator(Calculator):

    def substract(self, n1, n2):
        return n1 - n2

c = Calculator()
sc = SuperCalculator()
print(c.calculate(1, 2))
print(sc.calculate(1, 2))
print(sc.substract(10, 5))
# c.substract(10, 5) # Error
